src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  handleAddComment: (data: CommentData) => void;
+  postId: number;
+  isLoading: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  handleAddComment,
+  postId,
+  isLoading,
+}) => {
+  const [formData, setFormData] = useState({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [errors, setErrors] = useState({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newErrors = {
+      name: '',
+      email: '',
+      body: '',
+    };
+
+    if (!formData.name) {
+      newErrors.name = 'Name is required';
+    }
+
+    if (!formData.email) {
+      newErrors.email = 'Email is required';
+    }
+
+    if (!formData.body) {
+      newErrors.body = 'Enter some text';
+    }
+
+    setErrors(newErrors);
+
+    if (Object.values(newErrors).every((error) => error === '')) {
```

Comment:
no need to use `Object.values` and then `every`. it's not about performance
you can just check these value using logical `OR`